src/utils/get-sorted-nodes-by-import-order.ts

```diff
@@ -14,13 +12,15 @@ import { getSortedNodesGroup } from './get-sorted-nodes-group';
  * This function returns the given nodes, sorted in the order as indicated by
  * the importOrder array from the given options.
  * The plugin considers these import nodes as local import declarations.
- * @param nodes A subset of all import nodes that should be sorted.
+ * @param originalNodes A subset (of all import nodes) that should be sorted.
  * @param options Options to influence the behavior of the sorting algorithm.
  */
-export const getSortedNodesByImportOrder: GetSortedNodes = (nodes, options) => {
+export const getSortedNodesByImportOrder: GetSortedNodes = (
+    originalNodes,
+    options,
+) => {
     let { importOrder } = options;
 
-    const originalNodes = nodes.map(clone);
```

Comment:
Are we sure we don't need to clone?  It looks like the use of `getSortedNodesGroup` has a `.sort`, which will change them in place.  Though, I guess it's not mutating the nodes themselves, just their order within our `importOrderGroups` array.  So yeah, I think I've convinced myself this is fine.